Projects/18_Clustering/Predict_Sales.py

```diff
@@ -0,0 +1,36 @@
+# import data for offers
+import pandas as pd
+df_offers = pd.read_excel("./WineKMC.xlsx", sheetname=0)
+df_offers.columns = ["offer_id", "campaign", "varietal", "min_qty", "discount", "origin", "past_peak"]
+
+# import data for transactions
+df_transactions = pd.read_excel("./WineKMC.xlsx", sheetname=1)
+df_transactions.columns = ["customer_name", "offer_id"]
+df_transactions['n'] = 1
+
+# estimate total sales per offer_id
+df_total_sales = pd.merge(df_offers['offer_id'].to_frame(), df_transactions)
+df_total_sales = df_total_sales.groupby(['offer_id']).sum()
```

Comment:
You can do your aggregation by `offer_id` directly `offer_agg = df_transactions.groupby(['offer_id']).size()` and then merge with `df_offers` like this:
`df_offers.merge(offer_agg.to_frame(name='Total Sales'), left_index=True, right_index=True)`